gateway/sensor.go

```diff
@@ -307,6 +187,54 @@ func (g *Gateway) updateReadings(name string, newReadings map[string]interface{}
 
 	g.lastReadings[name] = readings
 }
+
+func (g *Gateway) DoCommand(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
+
+	// Validate that the dependency is correct.
+	if _, ok := cmd["validate"]; ok {
+		return map[string]interface{}{"validate": 1}, nil
+	}
+
+	// Add the nodes to the list of devices.
+	if newNode, ok := cmd["register_device"]; ok {
+		if newN, ok := newNode.(map[string]interface{}); ok {
+			node := convertToNode(newN)
+			g.devices[node.NodeName] = node
```

Comment:
any concern if a NodeName gets renamed? we would end up with dead/unresponsive nodes right?

Reply:
Added another do command to delete the node on close. Then a new one will be created with new name on rebuild.